### PR TITLE
Add kwarg types to Docker container run and create

### PR DESCRIPTION
**Pull request by adamtheturtle**
Implementations:

* `run`: https://github.com/docker/docker-py/blob/a3652028b1ead708bd9191efb286f909ba6c2a49/docker/models/containers.py#L534
* `create`: https://github.com/docker/docker-py/blob/a3652028b1ead708bd9191efb286f909ba6c2a49/docker/models/containers.py#L913

Notes:

`create`'s docstring says:

```
Takes the same arguments as :py:meth:`run`, except for ``stdout``,
``stderr``, and ``remove``.
```

However, the implementation includes:

```
kwargs['version'] = self.client.api._version
```

so I did not include `version` for `create`.

### CODE REVIEW

File: stubs/docker/docker/models/containers.pyi [hunk 1]
```diff
@@ -122,10 +209,182 @@ class ContainerCollection(Collection[Container]):
...
+        uts_mode: str | None = None,
+        version: str,
+        volume_driver: str | None = None,
+        volumes: dict[str, str] | list[str] | None = None,
+        volumes_from: list[str] | None = None,
+        working_dir: str | None = None,
+    ) -> Container: ...
+    def create(  # type:ignore[override]
+        self,
+        image: str | Image,
```
Comment (adamtheturtle) [inline stubs/docker/docker/models/containers.pyi:300]: I added `| Image` here.

File: stubs/docker/docker/_types.pyi [hunk 2]
```diff
@@ -1,8 +1,15 @@
...
+from pathlib import Path
+from typing import TypedDict, type_check_only
 from typing_extensions import TypeAlias
 
 # Type alias for JSON, explained at:
 # https://github.com/python/typing/issues/182#issuecomment-1320974824.
 JSON: TypeAlias = dict[str, JSON] | list[JSON] | str | int | float | bool | None
+
+@type_check_only
+class ContainerWeightDevice(TypedDict):
```
Comment (srittau) [inline stubs/docker/docker/_types.pyi:13]: See my comment on the other PR. We should prefix this with an underscore, to also make it clear that this is an internal class in places where this is used.

Comment (adamtheturtle) [reply stubs/docker/docker/_types.pyi:13]: Thank you. I will pause working on this PR until that other discussion is resolved.

**Review by srittau — APPROVED** [review]

File: stubs/docker/docker/models/containers.pyi [hunk 3]
```diff
@@ -113,9 +116,92 @@ class ContainerCollection(Collection[Container]):
...
+        tmpfs: dict[str, str] | None = None,
+        tty: bool = False,
+        ulimits: list[Ulimit] | None = None,
+        use_config_proxy: bool | None = None,
+        user: str | int | None = None,
+        userns_mode: str | None = None,
+        uts_mode: str | None = None,
+        version: str,
+        volume_driver: str | None = None,
+        volumes: dict[str, str] | list[str] | None = None,
```
Comment (nicocanali) [inline stubs/docker/docker/models/containers.pyi:201]: I think this annotation should be `dict[str, dict[str, str]]`.

According to the documentation

```
volumes (dict or list) –

A dictionary to configure volumes mounted inside the container. The key is either the host path or a volume name, and the value is a dictionary with the keys:

   - bind The path to mount the volume inside the container
   - mode Either rw to mount the volume read/write, or ro to mount it read-only.

For example:

{'/home/user1/': {'bind': '/mnt/vol2', 'mode': 'rw'},
 '/var/www': {'bind': '/mnt/vol1', 'mode': 'ro'}}
```

**Pull request merged**